### DISCUSSION TITLE
Understanding Around Provider id for Ride Hailing

### DISCUSSION THREAD

**Discussion by saurabhTriffy** [ONDC-Official/mobility-specification#60] [Q&A]
The parameter provider id is mandatory in onsearch and onwards. According to my understanding provider in case of ride hailing is the driver, who is also the agent.
We will have a provider in the system only after we assign a driver, which will be after confirm. So whats the significance of provider id before that? Shouldn't it be optional?

**Comment by 92shreyansh** [comment]
Considering Ride Hailing Service as a Item the provider us the one of this service online which can be the SellerApp however for this service the fulfillment of the service is performed by the driver hence agent.

This enables many future use cases or business models to function seemlessly without major changes in the model implementation.
